packages/protocol/contracts/dapis/RrpDapiServer.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "../AirnodeRrp.sol";
+import "./reducers/MeanMedianHybrid.sol";
+
+contract RrpDapiServer is MeanMedianHybrid {
+    struct Dapi {
+        uint256 noResponsesToReduce;
+        int256 toleranceInPercentages;
+        uint256 requesterIndex;
+        bytes32[] templateIds;
+        address[] designatedWallets;
+        address reduceAddress;
+        bytes4 reduceFunctionId;
+        mapping(uint256 => int256[]) requestIndexToResponses;
+        mapping(uint256 => uint256) requestIndexToResponsesLength;
+        uint64 nextRequestIndex;
+        uint64 requestIndexResetCount;
+        address requestIndexResetter;
```

Comment:
This is the "owner" of the dAPI, authorized to reset its request indices, which invalidates all previous requests. That didn't sound right when I wrote it.

Reply:
Yeah this will need a rework

Reply:
I'm not sure what you mean ... like, what happens to pending requests when the index is reset?

Reply:
Any fulfillments by Airnodes will revert. I thought about this a bit and maybe it's fine. Do a full review and let's have a call about it for real this time.